internal/core/encoding.go

```diff
@@ -288,3 +288,79 @@ func EncodeIndexDataStoreKey(key *IndexDataStoreKey) []byte {
 
 	return b
 }
+
+func DecodeDataStoreKey(data []byte) (DataStoreKey, error) {
+	if len(data) == 0 {
+		return DataStoreKey{}, ErrEmptyKey
+	}
+
+	if data[0] != '/' {
+		return DataStoreKey{}, ErrInvalidKey
+	}
+	data = data[1:]
+
+	data, colRootID, err := encoding.DecodeUvarintAscending(data)
+	if err != nil {
+		return DataStoreKey{}, err
+	}
+
+	var instanceType InstanceType
+	if len(data) > 1 {
+		if data[0] == '/' {
+			data = data[1:]
+		}
+		instanceType = InstanceType(data[0])
+		data = data[1:]
+	}
+
+	var docID string
+	if len(data) > 40 {
```

Comment:
todo: please extract the magic number to a const

Reply:
Will do

- [ ] Rm magic number